### PR TITLE
Initial commit of Django SDK

### PR DESCRIPTION
**Pull request by haosong**

### CODE REVIEW

File: wavefront_django_sdk/tracing.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+                scope.span.set_tag("error", "true")
+            scope.span.set_tag("span.kind", "server")
+            scope.span.set_tag("django.resource.module", module_name)
+            scope.span.set_tag("django.resource.func", func_name)
+            scope.span.set_tag("component", DJANGO_COMPONENT)
+            scope.span.set_tag("http.method", request.method)
+            scope.span.set_tag("http.url", request.build_absolute_uri())
+        scope.close()
+    #
+    # def _apply_tracing(self, request, view_func, attributes):
```
Comment (michael132) [inline wavefront_django_sdk/tracing.py]: lets remove these if not needed

File: requirements.txt [hunk 2]
```diff
@@ -0,0 +1,5 @@
+six
+Django>=1.10
+opentracing<2.0
+django_opentracing>=1.0
+wavefront_pyformance
```
Comment (ustinov) [inline requirements.txt]: Please add versions for `six` and `wavefront-pyformance` and sort the list alphabetically

File: wavefront_django_sdk/inject.py [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import six
+import opentracing
+
+
+def inject_as_headers(tracer, span, request):
```
Comment (ustinov) [inline wavefront_django_sdk/inject.py]: Docstrings missing

File: wavefront_django_sdk/middleware.py [hunk 4]
```diff
@@ -0,0 +1,353 @@
+import os
```
Comment (ustinov) [inline wavefront_django_sdk/middleware.py]: Sort imports alphabetically and you may add space between distinct groups to improve readability

File: wavefront_django_sdk/middleware.py [hunk 5]
```diff
@@ -0,0 +1,353 @@
...
+        func_name = resolve(request.path_info).func.__name__
+        module_name = resolve(request.path_info).func.__module__
+        self.update_gauge(
+            registry=self.reg,
+            key=self.get_metric_name(entity_name, request) + ".inflight",
+            tags=self.get_tags_map(
+                module_name=module_name,
+                func_name=func_name),
+            val=1
+        )
```
Comment (ustinov) [inline wavefront_django_sdk/middleware.py]: Be consistent in closing parens style, either keep it as you do on line 99 or wrap it everywhere 
https://www.python.org/dev/peps/pep-0008/#id16

File: wavefront_django_sdk/tracing.py [hunk 6]
```diff
@@ -0,0 +1,30 @@
+from django.urls import resolve
+from django_opentracing import tracing
+from wavefront_django_sdk.constants import DJANGO_COMPONENT
+
+
+class DjangoTracing(tracing.DjangoTracing):
```
Comment (ustinov) [inline wavefront_django_sdk/tracing.py]: Docstrings missing here and everywhere

File: setup.py [hunk 7]
```diff
@@ -0,0 +1,34 @@
+# coding: utf-8
```
[the remark this answers is not in the document]
Comment (song-hao) [reply setup.py]: https://github.com/wavefrontHQ/python-client/blob/master/setup.py#L1 from our other python repos.

**Comment by ustinov** [comment]
Please add the .travis.yml config as we did for the rest the rest of our other sdks
https://github.com/wavefrontHQ/wavefront-sdk-python/blob/master/.travis.yml
https://github.com/wavefrontHQ/wavefront-pyformance/blob/master/.travis.yml
https://github.com/wavefrontHQ/wavefront-opentracing-sdk-python/blob/master/.travis.yml

File: setup.py (hunk 7 above)
[the remark this answers is not in the document]
Comment (ustinov) [reply setup.py]: The above template is autogenerated by swagger-codegen and is far from perfection ;-(

Please follow the style of setup.py used by the rest of our other sdks
https://github.com/wavefrontHQ/wavefront-sdk-python/blob/master/setup.py
https://github.com/wavefrontHQ/wavefront-pyformance/blob/master/setup.py
https://github.com/wavefrontHQ/wavefront-opentracing-sdk-python/blob/master/setup.py

File: requirements.txt [hunk 8]
```diff
@@ -0,0 +1,5 @@
+six>=1.11
+Django>=1.10
```
Comment (ustinov) [inline requirements.txt]: Should we keep supporting the `1.10`?
The mainstream support has ended for it already https://www.djangoproject.com/download/#ft3

Comment (song-hao) [reply requirements.txt]: will update to `1.11`

File: .travis.yml [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+script:
+  - if [[ "${TRAVIS_PULL_REQUEST}" != "false" ]]; then
+      test -n "$(git diff -G version= $TRAVIS_COMMIT_RANGE setup.py)";
+    fi
+deploy:
+  - provider: pypi
+    user: wavefront-cs
+    server: https://test.pypi.org/legacy/
+    password:
+      secure: "w4j8TyINfzCRlcXiZcYj7eIcEYmYIisSHxUsXqC+7pUCFfAwOSJEQJhkYepFWxcmxOIhY5IdWOGsU1JNBkFyzS3NNlU76xvZ7Ykl5zicVd3ng6i/Pj6Eswhf2B5BgCpQwWIXAEHPQL+Q6xmY9tL4YnaLTpXsOOliltt1qdc+W31DilnCEmoFpDUK/KEQckRMvFj60mk3dsep87jxQdtuF+we0piN0JdG5KawjYbwbsLo940ZkXittRw5WKfcIwYbZo1SU8fxflvLKX48qbz6vK9+ACYL/3gav2x2zdopNx8mmK17Q5zgNKuvSm686gXZgzJGrNaSNFm3oNmRMP0dBS4ebYR+jS7ABOiT0SekwBqWkVKLEMnaJC/1k/8JuKabeXcA9mmZP/ZhmRsVMtyP449RBEQKoditMLr10PZbiz5iDu0UjqrF7yOqxwEsIUI7lBXY4r1FBxIrAD/xU8SMFurmb/qTmwB8sbmHxX/L6HBY0giiN039KqhAvcwPbSuDYchzyHouXRkhMgpI2jtpOVottEzMyJtRqaK+qV5zdm1ST6E7WqaIvEkkNM7OGw6jxexTadhQ7w4xdjHYIcgDHHFZJ3C8ZgivaiGaLNxrTT+OJ3fwc+zGcT6ndvYYGzXusre56Zal/PbdElv+sqX0sS/U5UVv9cJU5qQYIv3lpsI="
```
Comment (ustinov) [inline .travis.yml]: We need to generate a unique secret here, otherwise it won't work

Comment (song-hao) [reply .travis.yml]: Can you just make a commit to change? Or just send me the new secret.

File: wavefront_django_sdk/inject.py [hunk 10]
```diff
@@ -0,0 +1,18 @@
...
+import six
+
+
+# pylint: disable=protected-access
+def inject_as_headers(tracer, span, request):
+    """Inject tracing context into header."""
+    text_carrier = {}
+    tracer._tracer.inject(span.context, opentracing.Format.TEXT_MAP,
+                          text_carrier)
+    for key, val in six.iteritems(text_carrier):
```
Comment (ustinov) [inline wavefront_django_sdk/inject.py]: Just use `text_carrier.items()` instead and get rid of that `six` dependency

File: requirements.txt [hunk 11]
```diff
@@ -0,0 +1,5 @@
+opentracing>=2.0
```
Comment (ustinov) [inline requirements.txt]: Both `django-opentracing` and `wavefront-opentracing-sdk-python` 
have this dependency, do we need this one here, explicitly?

File: setup.py [hunk 12]
```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/env python3
```
Comment (ustinov) [inline setup.py]: Please do `chmod +x setup.py` in order to make this work

File: .travis.yml [hunk 13]
```diff
@@ -0,0 +1,38 @@
...
+    on:
+      python: 3.7
+  - provider: pypi
+    user: wavefront-cs
+    password:
+      secure: "w4j8TyINfzCRlcXiZcYj7eIcEYmYIisSHxUsXqC+7pUCFfAwOSJEQJhkYepFWxcmxOIhY5IdWOGsU1JNBkFyzS3NNlU76xvZ7Ykl5zicVd3ng6i/Pj6Eswhf2B5BgCpQwWIXAEHPQL+Q6xmY9tL4YnaLTpXsOOliltt1qdc+W31DilnCEmoFpDUK/KEQckRMvFj60mk3dsep87jxQdtuF+we0piN0JdG5KawjYbwbsLo940ZkXittRw5WKfcIwYbZo1SU8fxflvLKX48qbz6vK9+ACYL/3gav2x2zdopNx8mmK17Q5zgNKuvSm686gXZgzJGrNaSNFm3oNmRMP0dBS4ebYR+jS7ABOiT0SekwBqWkVKLEMnaJC/1k/8JuKabeXcA9mmZP/ZhmRsVMtyP449RBEQKoditMLr10PZbiz5iDu0UjqrF7yOqxwEsIUI7lBXY4r1FBxIrAD/xU8SMFurmb/qTmwB8sbmHxX/L6HBY0giiN039KqhAvcwPbSuDYchzyHouXRkhMgpI2jtpOVottEzMyJtRqaK+qV5zdm1ST6E7WqaIvEkkNM7OGw6jxexTadhQ7w4xdjHYIcgDHHFZJ3C8ZgivaiGaLNxrTT+OJ3fwc+zGcT6ndvYYGzXusre56Zal/PbdElv+sqX0sS/U5UVv9cJU5qQYIv3lpsI="
```
Comment (ustinov) [inline .travis.yml]:
```suggestion
      secure: "mm5C+7lnAlHwS1Wd8EIHbGj44qxweHEvyi4PNrhUOSo2Fd04m+EDXZHAXddKZiTdQSl0ZtlA4brlGYrx2CNBvNVKzfyWDnotQsfxlwSGZtvos10DSs1q3wtCAG2AVFBS6zdqza4qgy2GTMB8gdL6/8L1Yfht5nDSZDf6Z95QuoJgXuBTEmGTGIPmctUF/GTky/WXJv2X0appUxRoyypKNVjn3wkquEJIqddmEuH8fSF94qKwo2UNwg2HjCGGk6LiYA7WjoUIipxxpd1eQaHiogvdwUYj1CpvHWi+tENNHh2dmFd/pGRbgxxQ0K7NVLlIErafu41e/5wBaFL2qbfTFYK10B0Nv0F+gYhW/9/mRTS7ourJOWHWt5lOLPzDk9wpMU2U8oxeA5T5sal9MhKCAMWFNpiD8/Z8K0fYQDhTH5Vvh1zA6YSzwDvvZHCjuxzqlIccj1EwMsKnHTED9kbZDMok7rJ6Yv3e3GG9ULzFdgOcP5/Rk4POh36djD25E62emtghCmQE83uQg6iGsOVSCcmkAxnQJ4d/PvhuwpeqSFCjU3T76u70hST82u18Cp5O8DUwF+AyzQFx04Y2YQYsS2w589Mk58Qt6KVV+x+fTecfXfHlLJ33NbOftXFBlJZr5Kod7yM163FfwEOgq+FTqaEsn38Lo/8D+riGnKFjRrE="
```

File: .travis.yml [hunk 14]
```diff
@@ -0,0 +1,38 @@
...
+script:
+  - if [[ "${TRAVIS_PULL_REQUEST}" != "false" ]]; then
+      test -n "$(git diff -G version= $TRAVIS_COMMIT_RANGE setup.py)";
+    fi
+deploy:
+  - provider: pypi
+    user: wavefront-cs
+    server: https://test.pypi.org/legacy/
+    password:
+      secure: "w4j8TyINfzCRlcXiZcYj7eIcEYmYIisSHxUsXqC+7pUCFfAwOSJEQJhkYepFWxcmxOIhY5IdWOGsU1JNBkFyzS3NNlU76xvZ7Ykl5zicVd3ng6i/Pj6Eswhf2B5BgCpQwWIXAEHPQL+Q6xmY9tL4YnaLTpXsOOliltt1qdc+W31DilnCEmoFpDUK/KEQckRMvFj60mk3dsep87jxQdtuF+we0piN0JdG5KawjYbwbsLo940ZkXittRw5WKfcIwYbZo1SU8fxflvLKX48qbz6vK9+ACYL/3gav2x2zdopNx8mmK17Q5zgNKuvSm686gXZgzJGrNaSNFm3oNmRMP0dBS4ebYR+jS7ABOiT0SekwBqWkVKLEMnaJC/1k/8JuKabeXcA9mmZP/ZhmRsVMtyP449RBEQKoditMLr10PZbiz5iDu0UjqrF7yOqxwEsIUI7lBXY4r1FBxIrAD/xU8SMFurmb/qTmwB8sbmHxX/L6HBY0giiN039KqhAvcwPbSuDYchzyHouXRkhMgpI2jtpOVottEzMyJtRqaK+qV5zdm1ST6E7WqaIvEkkNM7OGw6jxexTadhQ7w4xdjHYIcgDHHFZJ3C8ZgivaiGaLNxrTT+OJ3fwc+zGcT6ndvYYGzXusre56Zal/PbdElv+sqX0sS/U5UVv9cJU5qQYIv3lpsI="
```
Comment (ustinov) [inline .travis.yml]:
```suggestion
      secure: "mm5C+7lnAlHwS1Wd8EIHbGj44qxweHEvyi4PNrhUOSo2Fd04m+EDXZHAXddKZiTdQSl0ZtlA4brlGYrx2CNBvNVKzfyWDnotQsfxlwSGZtvos10DSs1q3wtCAG2AVFBS6zdqza4qgy2GTMB8gdL6/8L1Yfht5nDSZDf6Z95QuoJgXuBTEmGTGIPmctUF/GTky/WXJv2X0appUxRoyypKNVjn3wkquEJIqddmEuH8fSF94qKwo2UNwg2HjCGGk6LiYA7WjoUIipxxpd1eQaHiogvdwUYj1CpvHWi+tENNHh2dmFd/pGRbgxxQ0K7NVLlIErafu41e/5wBaFL2qbfTFYK10B0Nv0F+gYhW/9/mRTS7ourJOWHWt5lOLPzDk9wpMU2U8oxeA5T5sal9MhKCAMWFNpiD8/Z8K0fYQDhTH5Vvh1zA6YSzwDvvZHCjuxzqlIccj1EwMsKnHTED9kbZDMok7rJ6Yv3e3GG9ULzFdgOcP5/Rk4POh36djD25E62emtghCmQE83uQg6iGsOVSCcmkAxnQJ4d/PvhuwpeqSFCjU3T76u70hST82u18Cp5O8DUwF+AyzQFx04Y2YQYsS2w589Mk58Qt6KVV+x+fTecfXfHlLJ33NbOftXFBlJZr5Kod7yM163FfwEOgq+FTqaEsn38Lo/8D+riGnKFjRrE="
```

File: wavefront_django_sdk/tracing.py [hunk 15]
```diff
@@ -0,0 +1,35 @@
...
+Wavefront Django Tracing.
+
+@author: Hao Song (songhao@vmware.com)
+"""
+
+from django.urls import resolve
+
+from django_opentracing import tracing
+
+from wavefront_django_sdk.constants import DJANGO_COMPONENT
```
Comment (ustinov) [inline wavefront_django_sdk/tracing.py]: You may use `from .constants import ...`

File: wavefront_django_sdk/middleware.py [hunk 16]
```diff
@@ -0,0 +1,433 @@
...
+from wavefront_pyformance.delta import delta_counter
+from wavefront_pyformance.tagged_registry import TaggedRegistry
+from wavefront_pyformance.wavefront_histogram import wavefront_histogram
+from wavefront_pyformance.wavefront_reporter import WavefrontReporter
+
+from wavefront_sdk.common import ApplicationTags, HeartbeaterService
+
+try:
+    # Django >= 1.10
+    from django.utils.deprecation import MiddlewareMixin
```
Comment (ustinov) [inline wavefront_django_sdk/middleware.py]: Do we need this if we don't support django older than `1.11`?

**Pull request merged**